specification.markdown

```diff
@@ -97,242 +170,298 @@ followed by the Data Size, and then the non-interpreted Binary itself:
-  </tr>
-  <tr class="version2">
-   <td>SignaturePublicKey<a name="SignaturePublicKey"></a></td>
-   <td>2+</td>
-   <td>[7E][A5]</td>
-   <td>-</td>
-   <td>-</td>
-   <td>-</td>
-   <td>-</td>
-   <td>binary</td>
-   <td>The public key to use with the algorithm (in the case of a PKI-based
-    signature).</td>
-  </tr>
-  <tr class="version2">
-   <td>Signature<a name="Signature"></a></td>
-   <td>2+</td>
-   <td>[7E][B5]</td>
-   <td>-</td>
-   <td>-</td>
-   <td>-</td>
-   <td>-</td>
-   <td>binary</td>
-   <td>The signature of the data (until a new.</td>
-  </tr>
-  <tr class="version2">
-   <td>SignatureElements<a name="SignatureElements"></a></td>
-   <td>2+</td>
-   <td>[7E][5B]</td>
-   <td>-</td>
-   <td>-</td>
-   <td>-</td>
-   <td>-</td>
-   <td>sub-elements</td>
-   <td>Contains elements that will be used to compute the signature.</td>
-  </tr>
-  <tr class="version2">
-   <td>SignatureElementList<a name="SignatureElementList"></a></td>
-   <td>3+</td>
-   <td>[7E][7B]</td>
-   <td>-</td>
-   <td>*</td>
-   <td>-</td>
-   <td>-</td>
-   <td>sub-elements</td>
-   <td>A list consists of a number of consecutive elements that represent one
-    case where data is used in signature. Ex: <i>Cluster|Block|BlockAdditional</i>
-    means that the BlockAdditional of all Blocks in all Clusters is used for
-    encryption.</td>
-  </tr>
-  <tr class="version2">
-   <td>SignedElement<a name="SignedElement"></a></td>
-   <td>4+</td>
-   <td>[65][32]</td>
-   <td>-</td>
-   <td>*</td>
-   <td>-</td>
-   <td>-</td>
-   <td>binary</td>
-   <td>An element ID whose data will be used to compute the signature.</td>
-  </tr>
- </tbody>
-</table>
+## Structure
+
+Syntactically an EBML document is a list of EBML elements. Each element
+has three parts; an Element ID, an Element Data Size and the element
+data payload. The element data payload is then either data, implicitly
+typed by the Element ID, or a list of EBML elements.
+
+EBML       = *ELEMENT
+ELEMENT    = ELEMENT_ID SIZE DATA
+DATA       = VALUE / *ELEMENT
+ELEMENT_ID = VINT
+SIZE       = VINT
+
+EBML uses big endian/network order byte order, i.e. most significant
+bit first. All of the tokens above are byte aligned.
+
+## Variable Size Integer
+
+For both Element ID and Element Data Size EBML uses a variable size
+integer, coded according to a schema similar to that of UTF-8 [UTF-8]
+encoding. The variable size integer begins with zero or more zero bits
+to define the width of the integer. Zero zeroes means a width of one
+byte, one zero a width of two bytes etc. The zeroes are followed by a
+marker of one set bit and then follows the actual integer value. The
+integer data consists of alignment data and tail data. The alignment
+data together with the width descriptor and the marker makes up one ore
+more complete bytes. The tail data is as many bytes as there were
+zeroes in the width descriptor, i.e. width-1.
+
+VINT           = VINT_WIDTH VINT_MARKER VINT_DATA
```

Comment:
Cleaned up in https://github.com/MediaArea/ebml-specification/commit/ca14fd2b9ec38cc4cae9b8730ff89a3f97160e6c